agent/consul/leader_connect_ca.go

```diff
@@ -760,6 +762,7 @@ func (c *CAManager) persistNewRootAndConfig(provider ca.Provider, newActiveRoot
 		Index:  idx,
 		Roots:  newRoots,
 		Config: &newConf,
+		Cas:    newConf.ModifyIndex,
```

Comment:
same here.
I set `cas` to ModifyIndex in here because we are reading modifying and writing the config in one place, I'm not sure about this may be I need to set it to 0 to avoid the check.